File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/typeconverters/IntegerListConverter.java
package com.me.simplehabit.persistence.typeconverters;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

public class IntegerListConverter {

    @TypeConverter
    public static String fromListToJson(List<Integer> list){
        return new Gson().toJson(list);
    }

    @TypeConverter
    public static List<Integer> fromJsonToList(String data){
        Type typelist=new TypeToken<List<Integer>>(){}.getType();

        return new Gson().fromJson(data,typelist);
    }
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/typeconverters/IntegerListConverter.kt
package com.me.simplehabit.persistence.typeconverters

import androidx.room.TypeConverter

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

import java.lang.reflect.Type

class IntegerListConverter {

    @TypeConverter
    fun fromListToJson(list: List<Int>): String {
        return Gson().toJson(list)
    }

    @TypeConverter
    fun fromJsonToList(data: String): List<Int>? {
        val typelist = object : TypeToken<List<Int>>() {

        }.type

        return Gson().fromJson<List<Int>>(data, typelist)
    }
}
